companion/gateway/security/semantic_injection_detector.py

```
@@ -0,0 +1,483 @@
+            # Use Groq API to analyze for threats with timeout
+            try:
+                response_text = await asyncio.wait_for(
+                    self.groq.chat_completion(
+                        messages=[{"role": "user", "content": prompt}],
+                        max_tokens=200,
+                        temperature=0.1  # Low temperature for consistent results
+                    ),
+                    timeout=5.0  # 5 second timeout for threat detection
+                )
+            except asyncio.TimeoutError:
+                self.logger.error(f"Threat detection timed out for user {user_id}")
```

Ruff (0.13.2)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)